### ISSUE TITLE
Use kss markup generated by sass loops.

### ISSUE THREAD

**Issue by jmdrawneek** [kss-node/kss-node#366]
I've been working on a project that generates colours and functions depending on some variables. Everything works great apart from kss node can't access the generated kss markup.

It ignores the output markup in the loops in the sass. And the css generated from sass removes the comments, unless prefixed and suffixed with /* */ which kss ignores.

This would be a great feature to have as it would cut down time writing kss markup. I'd also love to help create the feature if pointed in the right direction.

**Comment by jnowland** [comment]
You can't do loops in SASS inside of comments as they are commented out.

### but…
You can however generate a KSS document from a `.css` file.

The below code is how you can dynamically create a string inside of comments.

```
/*
Create KSS Color Pallet
*/

@function create-kss-colors($pallet, $pallet-modifier: '') {
  $class-list: '';

  @each $name, $color in map-get($colours, $pallet) {
    $class-list: '.Bg-#{$pallet-modifier}#{$name} - #{$name}'+'\A'+$class-list;
  }

  @return #{$class-list};
  
}

$kss-colors-ui: create-kss-colors(primary);

/*
Backgrounds

The various backgrounds and colours.

Markup: branding/swatches.hbs

#{$kss-colors}

Styleguide 1.backgrounds
*/
```

**Comment by jmdrawneek** [comment]
Thanks for the reply.

I'm actually doing this now. For some unknown reason kss node cli wasn't picking up the comments unless they were single line. So I'm using a gulp plugin now and it works fine. Although my modifiers don't seem to be converted now.

**Comment by jmdrawneek** [comment]
Oh and you can write comments in sass loops if they're multi line.

**Comment by jnowland** [comment]
Could you provide an example @jmdrawneek ?

**Comment by jmdrawneek** [comment]
// ----
// Sass (v3.4.21)
// Compass (v1.0.3)
// ----

Any ideas why the modifier in this don't get replaced?

`/*
Colours – grey

Markup:
<div class="{{modifier_class}}" style="height: 50px;">Test</div>

.grey-standard    - A smaller version of the default button.
.grey-dark        - A smaller version of the default button.
.grey-darker      - A smaller version of the default button.
.grey-darkest     - A smaller version of the default button.
.grey-light       - A smaller version of the default button.
.grey-lighter     - A smaller version of the default button.
.grey-lightest    - A smaller version of the default button.
.grey-bright      - A smaller version of the default button.
.grey-trans       - A smaller version of the default button.
.grey-complement  - A smaller version of the default button.

Styleguide 100.1.0.
*/`

They just get printed as {{modifier_class}} in kss.

`<td colspan="2"><div class="{{modifier_class}}" style="height: 50px;">Test</div></td>`

**Comment by jmdrawneek** [comment]
@jnowland Did you mean give an example of the commments in sass loops?

**Comment by jnowland** [comment]
@jmdrawneek yes the sass example.

**Comment by jmdrawneek** [comment]
This is a stipped down version of what I'm doing: 

`
$palette: (
        // Theme Palette colours.
        grey:   #666,
        blue:   #00F,
        red:    #F00,
        green:  #0F0,
);

@each $colour in $colours {
$i: $i + 1;
/*
Colours – #{$colour}

Markup:
<div class="{{modifier_class}}" style="height: 50px;">.{{modifier_class}}</div>

.#{$colour}-standard    - A smaller version of the default button.
.#{$colour}-dark        - A smaller version of the default button.
.#{$colour}-darker      - A smaller version of the default button.
.#{$colour}-darkest     - A smaller version of the default button.
.#{$colour}-light       - A smaller version of the default button.
.#{$colour}-lighter     - A smaller version of the default button.
.#{$colour}-lightest    - A smaller version of the default button.
.#{$colour}-bright      - A smaller version of the default button.
.#{$colour}-trans       - A smaller version of the default button.
.#{$colour}-complement  - A smaller version of the default button.

Styleguide 100.#{$i}.0.
*/
}`

**Comment by jmdrawneek** [comment]
@jnowland That now works and produces the kss markup/pages but the modifier doesn't get converted.

**Comment by jnowland** [comment]
@jmdrawneek So your putting the for each outside of the comment block - I'm not to sure is the the kss document pointing to a `.css` file after it renders it out?
I also noticed your markup is blank?

I did encounter some issues with the formatting of mine which messed up the kss rendering.

**Comment by jmdrawneek** [comment]
@jnowland Yep I'm using the css to build the kss stuff. My markup is blank?

**Comment by jnowland** [comment]
@jmdrawneek The markup being blank comment looks to be just a markdown formatting issue.

I always use external templates. Does this work if you manually code it out? If you can get it working maybe diff check the output in the CSS.

**Comment by jmdrawneek** [comment]
@jnowland Ok so it doesn't work if manually add it. I also tried:

https://github.com/kss-node/kss/blob/spec/SPEC.md#the-markup
(That documentation might want amending btw as the button tag isn't closed)

/*
Button

Your standard button suitable for clicking.

:hover    - Highlights when hovering.
:disabled - Dims the button when disabled.
.primary  - Indicates button is the primary action.
.smaller  - A little bit smaller now.

Markup:
<button class="button {{modifier_class}}"></button>

Styleguide 2.1.3
*/

**Comment by jmdrawneek** [comment]
@jnowland Very strange, tracked this all the way back to the section private function. Anyway, I though I'd give the master branch a go and that works fine. If there's anything I can help you guys with give me shout as I love the project and would be honored to help out.
